public/app/features/dashboard-scene/settings/variables/utils.ts

```diff
@@ -0,0 +1,112 @@
+import { SelectableValue } from '@grafana/data';
+import {
+  AdHocFiltersVariable,
+  ConstantVariable,
+  CustomVariable,
+  DataSourceVariable,
+  IntervalVariable,
+  TextBoxVariable,
+  QueryVariable,
+} from '@grafana/scenes';
+import { VariableType } from '@grafana/schema';
+
+import { AdHocFiltersVariableEditor } from './editors/AdHocFiltersVariableEditor';
+import { ConstantVariableEditor } from './editors/ConstantVariableEditor';
+import { CustomVariableEditor } from './editors/CustomVariableEditor';
+import { DataSourceVariableEditor } from './editors/DataSourceVariableEditor';
+import { IntervalVariableEditor } from './editors/IntervalVariableEditor';
+import { QueryVariableEditor } from './editors/QueryVariableEditor';
+import { TextBoxVariableEditor } from './editors/TextBoxVariableEditor';
+
+interface EditableVariableConfig {
+  name: string;
+  description: string;
+  // FIXME: This should include all the scene objects
+  scene:
+    | typeof CustomVariable
+    | typeof QueryVariable
+    | typeof ConstantVariable
+    | typeof IntervalVariable
+    | typeof DataSourceVariable
+    | typeof AdHocFiltersVariable
+    | typeof TextBoxVariable;
+  // FIXME: This should include all the editor objects
+  editor: React.ComponentType<any>;
+}
+
+export type EditableVariableType = Exclude<VariableType, 'system'>;
+
+export function isEditableVariableType(type: VariableType): type is EditableVariableType {
+  return type !== 'system';
+}
```

Comment:
not sure you need this as in scenes all the system variables are just macros and not part of the SceneVariableSet

Reply:
This exists because of the typecheck. Because `VariableType` was used around. Let me check if I can remove it. I have made plenty of changes since then. Good catch

Reply:
I checked and I needed it because `SceneVariable.state.type` is `VariableType`. So, we need to cast all the functionality for editing a variable.